SendFaceForRecognition/run.csx

```diff
@@ -1,7 +1,196 @@
+
+
+private const string FACE_API_SERVICE = "https://westeurope.api.cognitive.microsoft.com/face/v1.0/persongroups/dm-atmosphere";
+
+public static readonly string connectionString = CM.ConnectionStrings["Atmosphere"].ConnectionString;
+
+public const string CONTAINER_NAME_OUTPUT = "rectangles";
+
+public const int MAX_FACES_PER_PERSON = 248;
+
+public static async Task Run(string faceForRecognition, TraceWriter log)
+{
+    log.Info($"Triggered  - slackInput:{faceForRecognition}");
+    var slackInput = JsonConvert.DeserializeObject<SlackNotification>(faceForRecognition);
+    var imgUrl = $"{CM.AppSettings["images_endpoint"]}/{CONTAINER_NAME_OUTPUT}/{slackInput.FaceId}.jpg";
+
+    var totalCombined = await getCombinedVotes(slackInput, log);
+    var totalVotesPerUser = await getTotalVotesPerUser(slackInput, log);
+
+    if(totalCombined > 1 && totalVotesPerUser < MAX_FACES_PER_PERSON)
+    {  
+        var getedUsersMap = await getUsersMap(slackInput, log);
+        if(getedUsersMap!= null){
+            // Add face to person (face API) in Cognitive
+            log.Info($"Adding face to person.");
+            await addPersonFace(getedUsersMap, imgUrl, log);
+
+            // var getedPerson = await getPerson(getedUsersMap.CognitiveUid,log);
+            // log.Info($":::::::---getPerson---::::::::{getedPerson}");
+        }else{
+            // Create person in Cognitive
+            var createdPerson = await createPerson(slackInput.FaceUserId,log);
+
+            // Create usersMap in DB
+            var userMapping = new userMap();
+            userMapping.SlackUid = slackInput.FaceUserId;
+            userMapping.CognitiveUid = createdPerson;
+            await saveUsersMap(userMapping,log);
+
+            // Add face to person (face API) in Cognitive
+            await addPersonFace(userMapping, imgUrl, log);
+        }
+    }
+}
+
+private static async Task<string> createPerson(string slackUid, TraceWriter log)
+{
+    log.Info($"Create person in Cognitive with slackUid {slackUid}");
+
+    using (var client = new HttpClient())
+    using (var request = new HttpRequestMessage(HttpMethod.Post, FACE_API_SERVICE + "/persons"))
+    {
+        dynamic jsonObject = new JObject();
+        jsonObject.name = slackUid;
+        jsonObject.userData = slackUid;
+
+        request.Headers.Add("Ocp-Apim-Subscription-Key", CM.AppSettings["FaceServiceAPIKey"]);
+        byte[] byteData = Encoding.UTF8.GetBytes(jsonObject.ToString());
+        request.Content = new ByteArrayContent(byteData);
+        request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/json");
+
+        var response = await client.SendAsync(request);
+        var contents = await response.Content.ReadAsStringAsync();
+        return JsonConvert.DeserializeObject<personCreatedResponse>(contents).personId;
+    }
+}
+
+private static async Task<string> addPersonFace(userMap data, string image, TraceWriter log)
```

Comment:
Same comments as in `createPerson` and, also, methods should get only necessary parameters and not the whole object.